### PR TITLE
[320] Tableau de suivi déploiement

### PR DESCRIPTION
**Pull request by jdauphant**
image

### CODE REVIEW

**Comment by jdauphant** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
           

Coverage increased per file
===========================
+ app/helper/StringHelper.scala  100
         

Coverage decreased per file
===========================
- app/models/Area.scala  -2
- app/models/Organisation.scala  -16
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/aplus/aplus/pullRequest?prid=4726821&bid=15778950)

**Comment by lpereir4** [comment]
Est-ce que l'absence d'un lien pour accéder à la page de suivi est volontaire?

**Comment by jdauphant** [comment]
@lpereir4 non j'ai oublié de pousser tous les commits. Ça devrait être bon

**Comment by jdauphant** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/aplus/aplus/pullRequest?prid=4726821&bid=15778950)

**Comment by jdauphant** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aplus/aplus/pullRequest?prid=4726821&bid=15778950)

**Comment by jdauphant** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/aplus/aplus/pullRequest?prid=4726821&bid=15778950)

File: app/controllers/AreaController.scala [hunk 1]
```diff
@@ -46,4 +48,29 @@ class AreaController @Inject()(loginAction: LoginAction,
...
   }
+
+  def deploymentDashboard = loginAction {  implicit  request =>
+    asAdmin { () =>
+      "DEPLOYMENT_DASHBOARD_UNAUTHORIZED" -> s"Accès non autorisé au dashboard de déploiement"
+    } { () =>
+      val userGroups = userGroupService.allGroups
+      val users = userService.all
+
+      def usersIn(area: Area, organisation: Organisation): List[User] =
```
Comment (lpereir4) [inline app/controllers/AreaController.scala]: Je trouve le formatage ci-dessous plus lisible:
```
for {
  group <- userGroups.filter(group => group.areaIds.contains(area.id)
    && group.organisationSetOrDeducted.contains(organisation.shortName))
  user <- users if user.groupIds.contains(group.id)
} yield user

val data = for {
  area <- request.currentUser.areas.flatMap(Area.fromId)
} yield {
  val organisationMap: List[(Organisation, Int)] = for {
    organisation <- Organisation.all
    users = usersIn(area, organisation)
    userSum = users.count(_.instructor)
  } yield organisation -> userSum
  area -> organisationMap
}
```

**Pull request merged**